### PR TITLE
Add ability to spawn&resume application, compiler and rpc functionality

### PR DESCRIPTION
**Pull request by eybisi**

### CODE REVIEW

File: frida/src/compiler.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+use std::marker::PhantomData;
+use crate::{Frida, Error, error};
+
+/// Process management in Frida.
+pub struct Compiler<'a> {
+    compiler_ptr: *mut _FridaCompiler,
+    phantom: PhantomData<&'a _FridaCompiler>,
+}
+
+impl<'a> Compiler<'a> {
```
Comment (s1341) [inline frida/src/compiler.rs:12]: Why the lifetimes?

File: frida/src/compiler.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        'b: 'a,
+    {
+        Compiler {
+            compiler_ptr: unsafe { frida_sys::frida_compiler_new(frida_sys::frida_device_manager_new() ) },
+            phantom: PhantomData,
+        }
+    }
+
+    /// Build a frida script.
+    pub fn build(&self,program: &str) -> Result<&str,error::Error> {
```
Comment (s1341) [inline frida/src/compiler.rs:26]: please run `cargo fmt`

File: frida/src/compiler.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        let c_program_str = std::ffi::CString::new(program).map_err(|_| Error::CStringFailed);
+        //  gchar * frida_compiler_build_sync (FridaCompiler * self, const gchar * entrypoint, FridaBuildOptions * options, GCancellable * cancellable, GError ** error);
```
Comment (s1341) [inline frida/src/compiler.rs:28]: No need for this comment

File: frida/src/compiler.rs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        let mut error: *mut frida_sys::GError = std::ptr::null_mut();
+        let bundle = unsafe { frida_sys::frida_compiler_build_sync(
+                self.compiler_ptr,
+                c_program_str?.as_ptr() as _,
+                std::ptr::null_mut(),
```
Comment (s1341) [inline frida/src/compiler.rs:33]: We should probably allow specifying build options. Perhaps the `Compiler` struct should be a builder?

File: frida/src/device.rs [hunk 5]
```diff
@@ -119,6 +119,25 @@ impl<'a> Device<'a> {
             Err(Error::DeviceApplicationSpawnError)
         }
     }
+
+    /// Resumes the process with given pid.
+    pub fn resume(&self, pid: u32) -> Result<()> {
+        let mut error: *mut frida_sys::GError = std::ptr::null_mut();
+        unsafe {
+            // (FridaDevice * self, guint pid, GCancellable * cancellable, GError ** error);
```
Comment (s1341) [inline frida/src/device.rs:127]: Remove this comment.

File: frida/src/script.rs [hunk 6]
```diff
@@ -5,45 +5,136 @@
...
+#[derive(Deserialize, Debug)]
+pub struct SendMessage {
+    payload: Payload,
+}
+
+/// Represents a message payload.
+#[derive(Deserialize, Debug, Clone)]
+pub struct Payload {
+    /// Message type.
+    pub ty: String,
```
Comment (s1341) [inline frida/src/script.rs:64]: maybe make this `r#type`? `ty` is not very rusty.

Comment (eybisi) [reply frida/src/script.rs]: Changed it to r#type :+1:

**Comment by meme** [comment]
Please resolve the CI issues when possible. Thanks.

**Pull request closed without merging**